internal/image/validate.go

```diff
@@ -92,8 +93,42 @@ func ValidateImage(ctx context.Context, imageRef string, policy *ecc.EnterpriseC
 		return nil, err
 	}
 
+	now := time.Now()
+	// TODO: This should probably go somewhere else. Maybe in Evaluator.Evaluate?
+	for i, result := range results {
+		failures := []conftestOutput.Result{}
+		for _, failure := range result.Failures {
+			if !isEffective(failure, now) {
+				// TODO: Instead of moving to warnings, create new attribute: "futureViolations"
+				result.Warnings = append(result.Warnings, failure)
+			} else {
+				failures = append(failures, failure)
+			}
+		}
+		result.Failures = failures
+		results[i] = result
+	}
+
 	log.Debug("Conftest policy check complete")
 	out.SetPolicyCheck(results)
 
 	return out, nil
 }
+
+func isEffective(failure conftestOutput.Result, now time.Time) bool {
+	raw, ok := failure.Metadata["effective_on"]
```

Comment:
Perhaps also allow specifying metadata via command line parameter

Reply:
Good idea. I'll file a follow up story for this.

Reply:
https://issues.redhat.com/browse/HACBS-1219